bar.m

```diff
@@ -38,76 +38,30 @@
 
 #pragma mark - Support Functions and Classes
 
-// Voodoo magic to swizzle _crashOnException because it seems that touchbar is sensitive (esp to
-// system wakeups) and throws exceptions even when it really shouldn't... see discussion at
-// https://github.com/asmagill/hs._asm.undocumented.touchbar/issues/1 and
-// https://nshipster.com/method-swizzling/
-@interface NSApplication (crashInterceptor)
-+ (void)xxx_hammerspoon_SwizzleCrashOnException ;
-- (void)xxx_hammerspoon_crashOnException:(NSException *)exception ;
-@end
-
-@implementation NSApplication (crashInterceptor)
+@implementation NSFunctionRow: NSObject
 
-// it's a fair bet that we don't have to worry about a race condition with _crashOnException
-// because if some thread is executing this method, Hammerspoon is already crashing. While swizzling
-// in a method invoked when this module is loaded rather then when Hammerspoon is first loaded
-// is not ideal, I only want this work around installed when it's actually needed, i.e. when the
-// touchbar module is being used. It's a trade-off and I hope I've judged things correctly -- we'll
-// see...
-+ (void)xxx_hammerspoon_SwizzleCrashOnException {
++ (void)xxx_hammerspoon_SwizzleMarkActiveFunctionRowsAsDimmed {
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
         Class targetClass = [self class] ;
```

Comment:
And just to be complete for documentation purposes -- if you're trying to swizzle a class method, as opposed to an instance method, this line would be `Class targetClass = objc_getMetaClass(class_getName([self class])) ;` (and the invoking of the "original" method below would be `[NSFunctionRow xxx_hammerspoon_markActiveFunctionRowsAsDimmed:dimmed] ;`)

That said, I don't think it is, but figured I'd include this for completeness.

Reply:
Ok correction -- the double plusses screwed up my head -- you **are** trying to swizzle a class method. See the above comment.

Reply:
Ok again -- you'll need to decide which it is -- based on your class definition at the top, you define `markActiveFunctionRowsAsDimmed:` as a class method (+), but `xxx_hammerspoon_markActiveFunctionRowsAsDimmed:` as an instance method (-) -- they need to be the same, whichever is correct. And then use the correct `targetClass` as described in the first comment in this particular chain.